runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -1186,10 +1186,24 @@ class INTERPRETER_CLASS
 					_currentThread->tempSlot = (UDATA)invalidIndex;
 					rc = THROW_AIOB;
 				} else {
-					I_32 value = VM_ArrayCopyHelpers::referenceArrayCopy(_currentThread, srcObject, srcStart, destObject, destStart, elementCount);
-					if (-1 != value) {
-						buildInternalNativeStackFrame(REGISTER_ARGS);
-						rc = THROW_ARRAY_STORE;
+					I_32 errorIndex = VM_ArrayCopyHelpers::referenceArrayCopy(_currentThread, srcObject, srcStart, destObject, destStart, elementCount);
```

Comment:
I see the change here and in `FastJNI_java_lang_System.cpp` are duplicated. Can the change be made inside the VM helper `VM_ArrayCopyHelpers::referenceArrayCopy()` ?

Reply:
Good point, I would have to change the return value of `VM_ArrayCopyHelpers::referenceArrayCopy` to a `VM_BytecodeAction` but I suppose it would be worth it.